packages/api/src/record.ts

```diff
@@ -32,21 +65,16 @@ export class IPNSRecord {
         seqno: payload.seqno,
         validity: payload.validity
       }
-      await this.state.storage.put(data)
-      return jsonResponse(JSON.stringify(data), 200)
-    }
 
-    const map: Map<string, any> = await this.state.storage.get(['key', 'record', 'hasV2Sig', 'seqno', 'validity'])
+      if (record.key !== undefined && !canOverwrite(record, data)) {
```

Comment:
Is it possible to have a record without a key?

Reply:
Not very explicit but the `record.key !== undefined` is here to check that there is an existing record. No need to compare records if there is no prior.

When using:

`const map: Map<string, any> = await this.state.storage.get(['key', 'record', 'hasV2Sig', 'seqno', 'validity'])`

if there is no existing record then `key`, `record`, `hasV2Sig` etc... will be `undefined`.

This portion of code was improved in this PR: https://github.com/web3-storage/w3name/pull/14/files#diff-c18413c84e86453cd731abf8a6867ef9e5783e8a904aa5b2faeda3168bae2cc1R59